### PR TITLE
Feature: Migrator w/ CockroachDB support

### PR DESCRIPTION
**Pull request by tobbbles**
From https://github.com/LUSHDigital/microservice-core-golang/pull/15

### CODE REVIEW

**Comment by tobbbles** [comment]
This using `strings.Builder` means it requires go1.10+. Is this something we are happy about, or do we wish to support past Go versions?

File: migrate/migrate.go [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+	// Ensure a protocol exists. If it's gibberish it is not of our concern,
+	// as an error will be returned later when trying to connect via a
+	// gibberish protocol.
+	if len(co.protocol) == 0 {
+		return "", errors.New("microservice-db: unrecognised protocol for database connection")
+	}
+
+	// Write the protocol
+	conn.WriteString(co.protocol)
```
Comment (codingconcepts) [inline migrate/migrate.go:46]: Still think this bit could be done more concisely with a cheeky `fmt.Sprintf("%s://...")` :P

**Review by codingconcepts — APPROVED** [review]
LGTM mate.  Only point is the same point I made yesterday.  I think I was outgunned by everyone though :P

**Review by ladydascalie — CHANGES_REQUESTED** [review]
We need vendoring!

`dep ensure` this project and commit the vendor folder!

**Comment by tobbbles** [comment]
@ladydascalie Vendoring added via `dep`

**Review by ladydascalie — APPROVED** [review]

**Pull request merged**